### PR TITLE
Clarify README.md instructions for manual instrumentation

### PR DESCRIPTION
**Pull request by mpetazzoni**
See discussion #1945 for the background behind this change. The aim
is to make this repository's README easier to grok when trying to get
started instrumenting a Java application with OpenTelemetry.

### CODE REVIEW

**Comment by mpetazzoni** [comment]
I've been added to the CLA list by @flands. Hoping this comment will tickle the EasyCLA bot to re-evaluate?

**Comment by trask** [comment]
closing and re-opening to re-trigger CLA check(?)

**Comment by anuraaga** [comment]
@mpetazzoni I think you still need to click through to have your GitHub account associated, though not sure.

**Comment by mpetazzoni** [comment]
@anuraaga Thanks, it worked this time!

File: README.md [hunk 1]
```diff
@@ -79,7 +79,97 @@ java -javaagent:path/to/opentelemetry-javaagent-all.jar \
...
+    </dependency>
+  </dependencies>
+```
+
+#### Gradle
+
+```groovy
+dependencies {
+    compile('io.opentelemetry:opentelemetry-api:0.11.0')
+    compile('io.opentelemetry:opentelemetry-extension-annotations:0.11.0')
```
Comment (anuraaga) [inline README.md:114]:
```suggestion
    implementation('io.opentelemetry:opentelemetry-api:0.11.0')
    implementation('io.opentelemetry:opentelemetry-extension-annotations:0.11.0')
```

File: README.md [hunk 2]
```diff
@@ -79,7 +79,97 @@ java -javaagent:path/to/opentelemetry-javaagent-all.jar \
...
+}
+```
+
+#### Adding attributes to the current span
+
+A common need when instrumenting an application is to capture additional application-specific or
+business-specific information as additional attributes to an existing span from the automatic
+instrumentation. Grab the current span with `Span.current()` and use of the `setAttribute()`
```
Comment (anuraaga) [inline README.md:122]:
```suggestion
instrumentation. Grab the current span with `Span.current()` and use the `setAttribute()`
```

File: README.md [hunk 3]
```diff
@@ -79,7 +79,97 @@ java -javaagent:path/to/opentelemetry-javaagent-all.jar \
...
+```
+
+Each time the application invokes the annotated method, it creates a span that denote its duration
+and provides any thrown exceptions. Unless specified as an argument to the annotation, the span name
+will be `<className>.<methodName>`.
+
+##### Suppressing `@WithSpan` instrumentation
+
+Suppressing `@WithSpan` is useful if you have code that is over-instrumented using `@WithSpan`
+and you want to suppress some of them without modifying the code.
```
Comment (anuraaga) [inline README.md:157]: I know it's copied, but while we're updating the docs, for this, I'm not quite sure how this happens. `@WithSpan` is quite an explicit addition to code, how could it lead to over-instrumentation? @trask do you know the background of this quote? Would be nice if there's a real-world example here

Comment (trask) [reply README.md:157]: I do know the background! see https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/344#issuecomment-620818687

I don't think this section needs to be in top-level README, though, probably should be moved to https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/master/docs/suppressing-instrumentation.md

File: README.md [hunk 4]
```diff
@@ -79,7 +79,97 @@ java -javaagent:path/to/opentelemetry-javaagent-all.jar \
      -jar myapp.jar
 ```
 
-### Configuration parameters (subject to change!)
+### Manually instrumenting
```
Comment (trask) [inline README.md:82]: I'm not sure that the "Manually instrumenting" section should go above the "Configuration parameters" section?

Are you feeling like the README is getting too big, and stuff lower down is getting lost? Maybe we can split more things out into sub-pages and keep the main README lighter?

File: README.md (hunk 3 above)
Comment (mpetazzoni) [reply README.md:157]: I would imagine this happens if you're pulling in a dependency that uses `@WithSpan` and you need to suppress _their_ usage.

File: README.md (hunk 4 above)
Comment (mpetazzoni) [reply README.md:82]: Yes, that was the main issue I was trying to address here. If possible, I'd like to keep "restructuring" to another PR.

File: README.md (hunk 1 above)
Comment (mpetazzoni) [reply README.md:114]: I'm not sure I understand this change. The use of `compile()` was from the original README.

Comment (anuraaga) [reply README.md:114]: `compile` is deprecated and I think Gradle throws deprecation warnings now if using it. Our README was old, and if we keep it old now it'll stay old for a long time :P

File: README.md (hunk 3 above)
Comment (anuraaga) [reply README.md:157]: I think `OTEL_JAVAAGENT_EXCLUDE_CLASSES` is supposed to work for that case pretty well. So anyways, maybe I'll propose removing the flag but we don't have to do anything here :)

File: README.md (hunk 4 above)
Comment (trask) [reply README.md:82]: > If possible, I'd like to keep "restructuring" to another PR.

No problem. I'm still not sure about "Manual instrumentation" section going to the top though. My thinking is that "Manual instrumentation" section is generally something users need after they first get up and running with auto-instrumentation.

Comment (mpetazzoni) [reply README.md]: I've moved it back below and added a small sentence in the Getting Started section. Let me know if that works for you.

**Review by anuraaga — APPROVED** [review]

**Comment by mpetazzoni** [comment]
Thanks @anuraaga! Hopefully someone can merge before new conflicts appear.

**Comment by anuraaga** [comment]
@mpetazzoni We generally merge after two approvals, should be merged by tomorrow at the latest.

**Review by jkwatson — APPROVED** [review]
👍

**Review by trask — APPROVED** [review]

**Pull request merged**